src/main/java/com/faforever/client/status/StatPingService.java

```diff
@@ -0,0 +1,33 @@
+package com.faforever.client.status;
+
+import com.faforever.client.config.ClientProperties;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClient.Builder;
+import reactor.core.publisher.Flux;
+
+/** REST client for the statping-ng API. */
+@org.springframework.stereotype.Service
+@Lazy
+public class StatPingService {
+
+  private final WebClient webClient;
+
+  public StatPingService(ClientProperties clientProperties, Builder webClientBuilder) {
+    webClient = webClientBuilder
+        .baseUrl(clientProperties.getStatping().getApiRoot())
+        .build();
+  }
+
+  public Flux<Service> getServices() {
+    return getMany("/services", Service.class);
+  }
+
+  public Flux<Message> getMessages() {
+    return getMany("/messages", Message.class);
+  }
+
+  private <T> Flux<T> getMany(String path, Class<T> type) {
+    return webClient.get().uri(path).retrieve().bodyToFlux(type);
```

Comment:
It would probably be good to add error logging and properly return a Flux.empty() using onErrorResume

Reply:
Added it